src/app/modules/item/components/permissions-edit-dialog/permissions-edit-dialog.component.html

```diff
@@ -22,12 +22,16 @@
     <div class="dialog-content">
 
       <form [formGroup]="form">
-        <alg-collapsible-section i18n-title title="Can view" icon="fa fa-eye">
+        <alg-collapsible-section
+          i18n-title title="Can view" icon="fa fa-eye"
+          [collapsible]="!form.errors?.canView"
+        >
           <ng-template #content let-collapsed>
             <alg-progress-select
               [collapsed]="collapsed"
               formControlName="canView"
-              [values]="canViewValues"
+              [values]="progressSelectValues.canViewValues"
+              [validationErrors]="form.errors?.canView"
```

Comment:
Because the validation depends not on a single form control but on multiple (for example, the `can_grant_view` permission depends on the `can_view`), so the custom validator concerns the form group, and not individual form controls.

I tried to create a custom validator for each of the fields, but getting the value of adjacent fields (by getting the parent and then the form control of the other fields) but it seems like a bad practice, and it made the code more complex overall.

Reply:
I don't understand the link with validation, to me validation is defined at form-level and errors are stored into `form.errors` object, into properties that may match input names

Here, input name and error property seem to match, so according to the [doc](https://angular.io/api/forms/AbstractControl#errors) we should be able to also access the error from the `canView` FormControl via property `errors`